tests/CMakeLists.txt

```diff
@@ -182,12 +181,37 @@ function(add_cts_test)
   target_sources(test_all PRIVATE $<TARGET_OBJECTS:${test_exe_name}_objects>)
 endfunction()
 
-file(GLOB test_category_dirs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *)
-list(REMOVE_ITEM test_category_dirs "common")
-foreach(dir ${test_category_dirs})
-  if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/CMakeLists.txt")
-    add_subdirectory(${dir})
+function(add_cts_test)
+  # To make check that any .cpp files are passed
+  # List created because, direct check on "${ARGN}" gives false-positive result
+  set(tests_list "${ARGN}")
+  if (tests_list)
+    get_filename_component(test_dir ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+    set(test_exe_name ${test_dir})
+    set(test_cases_list "${ARGN}")
+
+    add_cts_test_helper(${test_exe_name} "${test_cases_list}")
   endif()
+endfunction()
+
+function(add_independent_cts_tests)
```

Comment:
Can you please add some comments to this function, what's the purpose of it. Additionally, please add comments to `add_cts_test` - I think the difference between them needs to be clearly communicated.

Reply:
comments was added in this commit: 78cae14b03d7ce5f688f024424fb5b91a0171a76